src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py

```diff
@@ -0,0 +1,1658 @@
+    def __init__(self, config: DebertaV2Config, **kwargs):
+        super().__init__(**kwargs)
+        self.dense = tf.keras.layers.Dense(config.pooler_hidden_size, name="dense")
+        self.dropout = TFDebertaV2StableDropout(config.pooler_dropout, name="dropout")
+        self.config = config
+
+    def call(self, hidden_states, training: bool = False):
+        # We "pool" the model by simply taking the hidden state corresponding
+        # to the first token.
+        context_token = hidden_states[:, 0]
+        context_token = self.dropout(context_token, training=training)
+        pooled_output = self.dense(context_token)
+        pooled_output = get_tf_activation(self.config.pooler_hidden_act)(pooled_output)
+        return pooled_output
+
+    @property
+    def output_dim(self) -> int:
+        return self.config.hidden_size
+
+
+class TFDebertaV2XSoftmax(tf.keras.layers.Layer):
+    """
+    Masked Softmax which is optimized for saving memory
+
+    Args:
+        input (:obj:`tf.Tensor`): The input tensor that will apply softmax.
+        mask (:obj:`tf.Tensor`): The mask matrix where 0 indicate that element will be ignored in the softmax calculation.
+        dim (int): The dimension that will apply softmax
+    """
+
+    def __init__(self, axis=-1, **kwargs):
+        super().__init__(**kwargs)
+        self.axis = axis
+
+    def call(self, inputs: tf.Tensor, mask: tf.Tensor):
+
+        rmask = tf.logical_not(tf.cast(mask, tf.bool))
+        output = tf.where(rmask, float("-inf"), inputs)
+        output = tf.nn.softmax(output, self.axis)
+        output = tf.where(rmask, 0.0, output)
+        return output
+
+
+def get_mask(input, dropout):
+    mask = tf.cast(
+        1 - tf.compat.v1.distributions.Bernoulli(probs=1 - dropout).sample(sample_shape=shape_list(input)), tf.bool
+    )
+    return mask, dropout
+
+
+@tf.custom_gradient
+def TFDebertaV2XDropout(input, local_ctx):
+    mask, dropout = get_mask(input, local_ctx)
+    scale = tf.convert_to_tensor(1.0 / (1 - dropout), dtype=tf.float32)
+    input = tf.cond(dropout > 0, lambda: tf.where(mask, 0.0, input) * scale, lambda: input)
+
+    def custom_grad(upstream_grad):
+        return tf.cond(
+            scale > 1, lambda: (tf.where(mask, 0.0, upstream_grad) * scale, None), lambda: (upstream_grad, None)
+        )
+
+    return input, custom_grad
+
+
+class TFDebertaV2StableDropout(tf.keras.layers.Layer):
+    """
+    Optimized dropout module for stabilizing the training
+
+    Args:
+        drop_prob (float): the dropout probabilities
+    """
+
+    def __init__(self, drop_prob, **kwargs):
+        super().__init__(**kwargs)
+        self.drop_prob = tf.convert_to_tensor(drop_prob, dtype=tf.float32)
+
+    def call(self, inputs: tf.Tensor, training: tf.Tensor = False):
+        if training and self.drop_prob > 0:
+            return TFDebertaV2XDropout(inputs, self.drop_prob)
+        return inputs
+
+
+class TFDebertaV2SelfOutput(tf.keras.layers.Layer):
```

Comment:
Same remark regarding the `# Copied from` statement (see below for the `TFDebertaV2Embeddings` comment for more details)

Reply:
done